misc/rot13.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file rot13.c
+ * @brief ROT13 is a simple letter substitution cipher that replaces a
+ * letter with the 13th letter after it in the alphabet.
+ * https://en.wikipedia.org/wiki/ROT13
+ * @details
```

Comment:
Please add algorithm details, how it works, etc.. You can take a bit of information from Wikipedia.